src/Arcus.Observability.Telemetry.Serilog.Enrichers/Extensions/LoggerEnrichmentConfigurationExtensions.cs

```diff
@@ -142,6 +143,7 @@ public static LoggerConfiguration WithVersion(this LoggerEnrichmentConfiguration
         /// <param name="transactionIdPropertyName">The name of the property to enrich the log event with the correlation transaction ID.</param>
         /// <exception cref="ArgumentNullException">Thrown when the <paramref name="enrichmentConfiguration"/> is <c>null</c>.</exception>
         /// <exception cref="ArgumentException">Thrown when the <paramref name="operationIdPropertyName"/> or <paramref name="transactionIdPropertyName"/> is blank.</exception>
+        [Obsolete("Use the " + nameof(WithCorrelationInfo) + " overload with providing your own default correlation accessor")]
```

Comment:
Hm why are we deprecating this? I don't want to have to provide a correlation info accessor every time myself, no?

Reply:
Well, it's static, so we will have the same correlation info accesor across multiple requests (possible with a different correlation information values). Which is not correct. We could return a new accessor on each get-call; but then we wouldn't be able to use that same accessor instance in the dependency container.

Reply:
As a consumer I should not have to care about all of this but just say that you want ot use correlation. So if that means we need to resolve the accessor through the DI container, than we should provide that as a parameter imo.

Reply:
Yes, you're right. I'll add an extension for this.

Reply:
You mean also deprecate the extension where you pass along the `ICorrelationInfoAccessor` yourself and guide them to the one with the `IServiceProvider`? Is there not some use cases where it would make sense to pass along the accessor yourself?

Reply:
Not really because with the approach where it has to be passed people would get the impression they have to do something while with the IServiceProvider we manage everything for them?